### PR TITLE
feat: steps support for circular progress bar

### PR DESCRIPTION
**Pull request by yykoypj**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/35802

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |steps support for circular progress bar           |
| 🇨🇳 Chinese |Progress: circle模式下支持steps           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 8070d5a</samp>

This pull request adds the feature of displaying a circular progress bar with steps or segments, using the `steps` prop in the `Progress` component. It also updates the documentation, the demo, and the tests for the `Progress` component to reflect this new feature. The main files affected are `components/progress/Circle.tsx`, `components/progress/progress.tsx`, and `components/progress/demo/circle-steps.tsx`.

### <samp>🤖 Generated by Copilot at 8070d5a</samp>

*  Add the `steps` prop to the `Progress` component for the `circle` and `dashboard` types, which allows the user to specify the total number of steps or the count and space of steps for the circular progress bar ([link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-8e2505ff307fc9096c36d6e175bde7ef8750d4caaef32c930e809b4483e7a4aaR32), [link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-8e2505ff307fc9096c36d6e175bde7ef8750d4caaef32c930e809b4483e7a4aaR55), [link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-8e2505ff307fc9096c36d6e175bde7ef8750d4caaef32c930e809b4483e7a4aaL66-R72), [link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL54-R54), [link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL164-R171))
*  Update the `strokeColor` prop for the `line` type, which can accept an array of colors for each step or segment of the progress bar, if the `steps` prop is defined ([link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-665865f861f3d35f0b0015cbda87052e798a0cac0c445644036750759eec1d6bL59-R60), [link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL134-R139))
*  Add two test cases for the `Progress` component with the `steps` prop for the `circle` and `dashboard` types, which check the rendering of the circular progress bar with steps using snapshots ([link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-c045f3b22eef7546211dddb79e6b4195b1e0a3dd6af6ae640c55579f9eeb5116R407-R416))
*  Add the demo code and documentation for the circular progress bar with steps, which show how to use the `steps` and `strokeColor` props for the `circle` and `dashboard` types, and update the version tags for the new props ([link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-be09355126aad10e535669fde2deb475cbdccec0bab2789f53ccd2cd2c2241f6R1-R7), [link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-11dc9a368d3395b2b72019300e7837ff1d2553e16f1994419704feca9c9420f7R1-R26), [link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-665865f861f3d35f0b0015cbda87052e798a0cac0c445644036750759eec1d6bR36), [link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-665865f861f3d35f0b0015cbda87052e798a0cac0c445644036750759eec1d6bR67), [link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-665865f861f3d35f0b0015cbda87052e798a0cac0c445644036750759eec1d6bL72-R75), [link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-bc17738f93f52688339916ab8d5ca2963cbef403a143b26a40743c1a83dbd2ffR37), [link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-bc17738f93f52688339916ab8d5ca2963cbef403a143b26a40743c1a83dbd2ffL54), [link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-bc17738f93f52688339916ab8d5ca2963cbef403a143b26a40743c1a83dbd2ffR67), [link](https://github.com/ant-design/ant-design/pull/42821/files?diff=unified&w=0#diff-bc17738f93f52688339916ab8d5ca2963cbef403a143b26a40743c1a83dbd2ffR75))

### CODE REVIEW

File: components/progress/index.en-US.md [hunk 1]
```diff
@@ -63,13 +64,15 @@ Properties that shared by all types.
 
 | Property | Description | Type | Default | Version |
 | --- | --- | --- | --- | --- |
+| steps | The total step count | number \| { count: number, space: number } | - | 5.5.3 |
 | strokeColor | The color of circular progress, render `linear-gradient` when passing an object | string \| object | - | - |
 | strokeWidth | To set the width of the circular progress, unit: percentage of the canvas width | number | 6 | - |
 
 ### `type="dashboard"`
 
 | Property | Description | Type | Default |
```
Comment (kiner-tang) [inline components/progress/index.en-US.md:73]: 表头也补一下 Version 吧

File: components/progress/index.en-US.md [hunk 2]
```diff
@@ -63,13 +64,15 @@ Properties that shared by all types.
...
-| --- | --- | --- | --- |
+| --- | --- | --- | --- | --- |
+| steps | The total step count | number \| { count: number, space: number } | - | 5.5.3 |
```
Comment (kiner-tang) [inline components/progress/index.en-US.md:75]:
```suggestion
| steps | The total step count | number \| { count: number, space: number } | - | 5.5.3, `{ count: number, space: number }`: 5.6.0 |
```

File: components/progress/index.zh-CN.md [hunk 3]
```diff
@@ -51,7 +52,6 @@ demo:
 | success | 成功进度条相关配置 | { percent: number, strokeColor: string } | - | - |
 | trailColor | 未完成的分段的颜色 | string | - | - |
 | type | 类型，可选 `line` `circle` `dashboard` | string | `line` | - |
-| size | 进度条的尺寸 | number \| \[number, number] \| "small" \| "default" | "default" | v5.3.0 |
```
Comment (kiner-tang) [inline components/progress/index.zh-CN.md:54]: 这个为什么要删除

File: components/progress/index.zh-CN.md [hunk 4]
```diff
@@ -64,13 +64,15 @@ demo:
 
 | 属性 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
+| steps | 进度条总共步数 | number \| { count: number, space: number } | - | 5.5.3 |
```
Comment (kiner-tang) [inline components/progress/index.zh-CN.md:67]: 跟英文文档一样，改一下版本

**Comment by yykoypj** [comment]
@kiner-tang 
steps原本在dashboard模式和circle模式下并不支持, number类型和object类型是同时在这里支持的。

**Comment by kiner-tang** [comment]
> @kiner-tang steps原本在dashboard模式和circle模式下并不支持, number类型和object类型是同时在这里支持的。

如果原本没有 steps 属性，这次新增的支持的话，那就整个直接标记版本 5.6.0，代表 5.6.0 之后才支持。

File: components/progress/progress.tsx [hunk 5]
```diff
@@ -51,7 +51,7 @@ export interface ProgressProps extends ProgressAriaProps {
   gapDegree?: number;
   gapPosition?: 'top' | 'bottom' | 'left' | 'right';
   size?: number | [number, number] | ProgressSize;
-  steps?: number;
+  steps?: number | { count: number; space: number };
```
Comment (kiner-tang) [inline components/progress/progress.tsx:54]: space 有什么用，好像没看到哪里有取用的。

File: components/progress/index.zh-CN.md [hunk 6]
```diff
@@ -64,13 +65,15 @@ demo:
 
 | 属性 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
+| steps | 进度条总共步数 | number \| { count: number, space: number } | - | 5.6.0 |
```
Comment (kiner-tang) [inline components/progress/index.zh-CN.md:68]: 稍微说明一下 `count` 和 `space` 都是拿来做什么的

File: components/progress/progress.tsx (hunk 5 above)
Comment (kiner-tang) [reply components/progress/progress.tsx:54]: space 感觉可以改成可选的，然后提供一个默认值，改 space 的场景应该不多，大多数情况用默认的就好

File: components/progress/index.en-US.md [hunk 7]
```diff
@@ -64,15 +64,15 @@ Properties that shared by all types.
...
 | strokeColor | The color of circular progress, render `linear-gradient` when passing an object | string \| object | - | - |
 | strokeWidth | To set the width of the circular progress, unit: percentage of the canvas width | number | 6 | - |
 
 ### `type="dashboard"`
 
-| Property | Description | Type | Default |
+| Property | Description | Type | Default | Version |
 | --- | --- | --- | --- | --- |
-| steps | The total step count | number \| { count: number, space: number } | - | 5.6.0 |
+| steps | The total step count.When passing an object, `count` refers to the number of steps, and `space` refers to the distance between them.When passing number, the default value for `space` is 2. | number \| { count: number, space: number } | - | 5.6.0 |
```
Comment (afc163) [inline components/progress/index.en-US.md:75]:
```suggestion
| steps | The total step count. When passing an object, `count` refers to the number of steps, and `space` refers to the distance between them. When passing number, the default value for `space` is 2. | number \| { count: number, space: number } | - | 5.6.0 |
```

**Comment by afc163** [comment]
冲突了

**Comment by afc163** [comment]
冲突了，ping

**Comment by afc163** [comment]
测试用例没过

**Comment by yykoypj** [comment]
大佬~这个argos是个啥，怎么排查

**Comment by yykoypj** [comment]
和最新的代码有冲突耶，需要resolve不

**Comment by afc163** [comment]
冲突了，解决一下吧。

**Comment by afc163** [comment]
Hello~ @yykoypj

**Comment by yykoypj** [comment]
> Hello~ @yykoypj

收到，稍晚哈

**Comment by yykoypj** [comment]
新的pr地址：https://github.com/ant-design/ant-design/pull/47940

**Pull request closed without merging**